File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/profile/GetUserProfile.java
package com.moovie.backend.domain.usecase.profile;

import com.moovie.backend.domain.model.UserProfile;
import com.moovie.backend.domain.repository.ProfileRepository;
import org.springframework.stereotype.Service;

@Service
public class GetUserProfile {

    private final ProfileRepository profileRepository;

    public GetUserProfile(ProfileRepository profileRepository) {
        this.profileRepository = profileRepository;
    }

    public UserProfile invoke() {
        return profileRepository.getUserProfile();
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/profile/GetUserProfile.kt
package org.mobyle.domain.usecase.profile

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.model.UserProfile
import org.mobyle.domain.repository.ProfileRepository

class GetUserProfile(private val repository: ProfileRepository) {
    operator fun invoke(): UserProfile = runBlocking {
        repository.getUserProfile()
    }
}
